In [1]:
import pandas as pd
import openpyxl as opx
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
teisdf = pd.read_excel('../data/teisdata.xlsx', 
                             sheet_name=['ECO with Exit21-22', 'Elig Timeline Rpt 2018-2022'])

ecodf = teisdf.get('ECO with Exit21-22')
eligdf = teisdf.get('Elig Timeline Rpt 2018-2022')
ecodf.head()

DISTRICT  CHILD_ID ECO_Entry_DATE  Ent SOCIAL_SCALE  Ent KNOWLEDGE_SCALE  \
0       ET  500335.0     2021-04-23               5.0                  3.0   
1       ET  479453.0     2020-03-23               7.0                  7.0   
2       ET  510663.0            NaT               NaN                  NaN   
3       ET  452482.0     2019-01-31               2.0                  3.0   
4       ET  506507.0            NaT               NaN                  NaN   

   Ent APPROPRIATE_ACTION_SCALE ECO_Exit_DATE  Exit SOCIAL_SCALE  \
0                           4.0    2021-10-23                6.0   
1                           7.0    2021-08-19                7.0   
2                           NaN           NaT                NaN   
3                           3.0    2021-11-22                5.0   
4                           NaN           NaT                NaN   

   Exit KNOWLEDGE_SCALE  Exit APPROPRIATE_ACTION_SCALE  ... OC2 Data Check  \
0                   3.0                            4.0  ...            1.0   
1                   7.0                            7.0  ...            1.0   
2                   NaN                            NaN  ...            0.0   
3                   3.0                            5.0  ...            1.0   
4                   NaN                            NaN  ...            0.0   

  OC3 - a OC3 - b OC3 - c OC3 - d OC3 - e OC3 Data Check  \
0     NaN     1.0     NaN     NaN     NaN            1.0   
1     NaN     NaN     NaN     NaN     1.0            1.0   
2     NaN     NaN     NaN     NaN     NaN            0.0   
3     NaN     NaN     1.0     NaN     NaN            1.0   
4     NaN     NaN     NaN     NaN     NaN            0.0   

   Days btw Initial and Exit  Days btw I-IFSP to Exit ECO  \
0                      206.0                        183.0   
1                      704.0                        514.0   
2                      115.0                     -44483.0   
3                     1053.0                       1026.0   
4                      159.0                     -44434.0   

   Days btw I-IFSP to entry ECO  
0                           0.0  
1                           0.0  
2                      -44483.0  
3                           0.0  
4                      -44434.0  

[5 rows x 60 columns]

In [3]:
eligdf.head()

District  Child ID Child Status  Child Phase Init. IFSP Due Date  \
0       ET    453926     Inactive         IFSP          2019-03-04   
1       ET    431729     Inactive         IFSP          2018-10-26   
2       ET    462474     Inactive         IFSP          2019-06-28   
3       ET    446841     Inactive  Eligibility          2018-11-23   
4       ET    459629     Inactive         IFSP          2019-05-16   

  IFSP Due Date IFSP Late Reason Active Ref. Date Parent Consent/Intake Date  \
0    2019-03-27           System       2019-01-18                 2019-02-04   
1    2018-10-26              NaN       2018-09-11                 2018-09-25   
2    2019-06-14              NaN       2019-05-14                 2019-05-30   
3           NaT              NaN       2018-10-09                 2018-10-25   
4    2019-04-24              NaN       2019-04-01                 2019-04-10   

  Date Dev. Evaluator Rec'd. Ref.  ... # Days Parent Consent to Dev. Eval.  \
0             2019-01-22 00:00:00  ...                                17.0   
1             2018-09-12 00:00:00  ...                                 0.0   
2             2019-05-15 00:00:00  ...                                 0.0   
3             2018-10-11 00:00:00  ...                                 0.0   
4             2019-04-02 00:00:00  ...                                 0.0   

  # Days Dev. Evaluator Assigned to Dev. Eval.  \
0                                         30.0   
1                                         13.0   
2                                         15.0   
3                                         14.0   
4                                          8.0   

  1st Date Requested to Receipt of Med. Records  \
0                                           NaN   
1                                           NaN   
2                                           0.0   
3                                           NaN   
4                                           0.0   

  # Days Dev. Eval. to Elig. Det. # Days Ref. to Elig. Det.  \
0                             4.0                      38.0   
1                             6.0                      20.0   
2                             1.0                      17.0   
3                             1.0                      17.0   
4                             2.0                      11.0   

  Init. Elig. Decision                           Init. Elig. Category  \
0             Eligible                       Developmental Evaluation   
1             Eligible  Diagnosed Condition, Developmental Evaluation   
2             Eligible                       Developmental Evaluation   
3             Eligible                            Diagnosed Condition   
4             Eligible                       Developmental Evaluation   

  Init. Elig. Det. Date  Exit Date                              Exit Reason  
0            2019-02-25 2020-01-11                    618 - Part B eligible  
1            2018-10-01 2020-12-11  618 - Part B eligibility not determined  
2            2019-05-31 2020-11-12  618 - Part B eligibility not determined  
3            2018-10-26 2020-08-12                           Parent decline  
4            2019-04-12 2019-09-26                    618 - Parent withdraw  

[5 rows x 38 columns]

In [4]:
ecodf.tail()

DISTRICT  CHILD_ID  \
8631                                                 UC  514216.0   
8632                                                NaN       NaN   
8633                                                NaN       NaN   
8634                                                NaN       NaN   
8635  9/29/22 JM Corrected in TEIDS. Progress would ...       NaN   

     ECO_Entry_DATE  Ent SOCIAL_SCALE  Ent KNOWLEDGE_SCALE  \
8631            NaT               NaN                  NaN   
8632            NaT               NaN                  NaN   
8633            NaT               NaN                  NaN   
8634            NaT               NaN                  NaN   
8635            NaT               NaN                  NaN   

      Ent APPROPRIATE_ACTION_SCALE ECO_Exit_DATE  Exit SOCIAL_SCALE  \
8631                           NaN           NaT                NaN   
8632                           NaN           NaT                NaN   
8633                           NaN           NaT                NaN   
8634                           NaN           NaT                NaN   
8635                           NaN           NaT                NaN   

      Exit KNOWLEDGE_SCALE  Exit APPROPRIATE_ACTION_SCALE  ... OC2 Data Check  \
8631                   NaN                            NaN  ...            0.0   
8632                   NaN                            NaN  ...            NaN   
8633                   NaN                            NaN  ...            NaN   
8634                   NaN                            NaN  ...            NaN   
8635                   NaN                            NaN  ...            NaN   

     OC3 - a OC3 - b OC3 - c OC3 - d OC3 - e OC3 Data Check  \
8631     NaN     NaN     NaN     NaN     NaN            0.0   
8632     NaN     NaN     NaN     NaN     NaN            NaN   
8633     NaN     NaN     NaN     NaN     NaN            NaN   
8634     NaN     NaN     NaN     NaN     NaN            NaN   
8635     NaN     NaN     NaN     NaN     NaN            NaN   

      Days btw Initial and Exit  Days btw I-IFSP to Exit ECO  \
8631                       65.0                     -44537.0   
8632                        NaN                          NaN   
8633                        NaN                          NaN   
8634                        NaN                          NaN   
8635                        NaN                          NaN   

      Days btw I-IFSP to entry ECO  
8631                      -44537.0  
8632                           NaN  
8633                           NaN  
8634                           NaN  
8635                           NaN  

[5 rows x 60 columns]

In [5]:
eligdf.tail()

District  Child ID Child Status  Child Phase Init. IFSP Due Date  \
67605       UC    509553     Inactive         IFSP          2021-10-03   
67606       UC    509684     Inactive         IFSP          2021-10-07   
67607       UC    526338     Inactive         IFSP          2022-05-22   
67608       UC    511641     Inactive  Eligibility          2021-11-22   
67609       UC    519977     Inactive         IFSP          2022-03-07   

      IFSP Due Date IFSP Late Reason Active Ref. Date  \
67605    2021-10-22           System       2021-08-19   
67606    2021-09-07              NaN       2021-08-23   
67607    2022-05-05              NaN       2022-04-07   
67608           NaT              NaN       2021-10-08   
67609    2022-02-03              NaN       2022-01-21   

      Parent Consent/Intake Date Date Dev. Evaluator Rec'd. Ref.  ...  \
67605                 2021-08-26             2021-08-23 00:00:00  ...   
67606                 2021-08-31             2021-08-24 00:00:00  ...   
67607                 2022-04-10             2022-04-08 00:00:00  ...   
67608                 2021-10-18             2021-10-11 00:00:00  ...   
67609                 2022-01-26             2022-01-24 00:00:00  ...   

      # Days Parent Consent to Dev. Eval.  \
67605                                 0.0   
67606                                 0.0   
67607                                 1.0   
67608                                 0.0   
67609                                 0.0   

      # Days Dev. Evaluator Assigned to Dev. Eval.  \
67605                                          3.0   
67606                                          7.0   
67607                                          3.0   
67608                                          7.0   
67609                                          2.0   

      1st Date Requested to Receipt of Med. Records  \
67605                                          35.0   
67606                                           0.0   
67607                                           0.0   
67608                                           NaN   
67609                                           0.0   

      # Days Dev. Eval. to Elig. Det. # Days Ref. to Elig. Det.  \
67605                            47.0                      54.0   
67606                             2.0                      10.0   
67607                             2.0                       6.0   
67608                            11.0                      21.0   
67609                             5.0                      10.0   

      Init. Elig. Decision                           Init. Elig. Category  \
67605             Eligible  Diagnosed Condition, Developmental Evaluation   
67606             Eligible                       Developmental Evaluation   
67607             Eligible                       Developmental Evaluation   
67608             Eligible                            Diagnosed Condition   
67609             Eligible                       Developmental Evaluation   

      Init. Elig. Det. Date  Exit Date               Exit Reason  
67605            2021-10-12 2021-12-10  618 - Moved out of state  
67606            2021-09-02 2022-03-07     618 - Parent withdraw  
67607            2022-04-13 2022-07-25     618 - Parent withdraw  
67608            2021-10-29 2022-06-22            Parent decline  
67609            2022-01-31 2022-06-20     618 - Part B eligible  

[5 rows x 38 columns]

In [6]:
ecodf = ecodf.iloc[:-4 , :]

In [7]:
ecodf.tail()

DISTRICT  CHILD_ID ECO_Entry_DATE  Ent SOCIAL_SCALE  Ent KNOWLEDGE_SCALE  \
8627       UC  485978.0     2020-08-19               1.0                  3.0   
8628       UC  488802.0     2020-09-21               4.0                  4.0   
8629       UC  516136.0            NaT               NaN                  NaN   
8630       UC  490688.0     2020-10-27               4.0                  5.0   
8631       UC  514216.0            NaT               NaN                  NaN   

      Ent APPROPRIATE_ACTION_SCALE ECO_Exit_DATE  Exit SOCIAL_SCALE  \
8627                           5.0    2021-02-11                5.0   
8628                           3.0    2021-09-09                6.0   
8629                           NaN           NaT                NaN   
8630                           7.0    2021-10-13                7.0   
8631                           NaN           NaT                NaN   

      Exit KNOWLEDGE_SCALE  Exit APPROPRIATE_ACTION_SCALE  ... OC2 Data Check  \
8627                   7.0                            7.0  ...            1.0   
8628                   7.0                            7.0  ...            1.0   
8629                   NaN                            NaN  ...            0.0   
8630                   7.0                            7.0  ...            1.0   
8631                   NaN                            NaN  ...            0.0   

     OC3 - a OC3 - b OC3 - c OC3 - d OC3 - e OC3 Data Check  \
8627     NaN     NaN     NaN     1.0     NaN            1.0   
8628     NaN     NaN     NaN     1.0     NaN            1.0   
8629     NaN     NaN     NaN     NaN     NaN            0.0   
8630     NaN     NaN     NaN     NaN     1.0            1.0   
8631     NaN     NaN     NaN     NaN     NaN            0.0   

      Days btw Initial and Exit  Days btw I-IFSP to Exit ECO  \
8627                      361.0                        176.0   
8628                      353.0                        353.0   
8629                      130.0                     -44574.0   
8630                      495.0                        351.0   
8631                       65.0                     -44537.0   

      Days btw I-IFSP to entry ECO  
8627                           0.0  
8628                           0.0  
8629                      -44574.0  
8630                           0.0  
8631                      -44537.0  

[5 rows x 60 columns]

In [8]:
dedf = eligdf[eligdf['Init. Elig. Category'] == 'Developmental Evaluation']

In [9]:
dedf.head()

District  Child ID Child Status Child Phase Init. IFSP Due Date  \
0       ET    453926     Inactive        IFSP          2019-03-04   
2       ET    462474     Inactive        IFSP          2019-06-28   
4       ET    459629     Inactive        IFSP          2019-05-16   
5       ET    446765     Inactive        IFSP          2019-06-30   
6       ET    437348     Inactive        IFSP          2019-03-08   

  IFSP Due Date IFSP Late Reason Active Ref. Date Parent Consent/Intake Date  \
0    2019-03-27           System       2019-01-18                 2019-02-04   
2    2019-06-14              NaN       2019-05-14                 2019-05-30   
4    2019-04-24              NaN       2019-04-01                 2019-04-10   
5    2019-06-05              NaN       2019-05-16                 2019-05-24   
6    2019-02-11              NaN       2019-01-22                 2019-02-11   

  Date Dev. Evaluator Rec'd. Ref.  ... # Days Parent Consent to Dev. Eval.  \
0             2019-01-22 00:00:00  ...                                17.0   
2             2019-05-15 00:00:00  ...                                 0.0   
4             2019-04-02 00:00:00  ...                                 0.0   
5             2019-05-17 00:00:00  ...                                 0.0   
6                             NaN  ...                              -255.0   

  # Days Dev. Evaluator Assigned to Dev. Eval.  \
0                                         30.0   
2                                         15.0   
4                                          8.0   
5                                          8.0   
6                                          NaN   

  1st Date Requested to Receipt of Med. Records  \
0                                           NaN   
2                                           0.0   
4                                           0.0   
5                                           0.0   
6                                           NaN   

  # Days Dev. Eval. to Elig. Det. # Days Ref. to Elig. Det.  \
0                             4.0                      38.0   
2                             1.0                      17.0   
4                             2.0                      11.0   
5                             4.0                      12.0   
6                             7.0                    -228.0   

  Init. Elig. Decision      Init. Elig. Category Init. Elig. Det. Date  \
0             Eligible  Developmental Evaluation            2019-02-25   
2             Eligible  Developmental Evaluation            2019-05-31   
4             Eligible  Developmental Evaluation            2019-04-12   
5             Eligible  Developmental Evaluation            2019-05-28   
6             Eligible  Developmental Evaluation            2018-06-08   

   Exit Date                                        Exit Reason  
0 2020-01-11                              618 - Part B eligible  
2 2020-11-12            618 - Part B eligibility not determined  
4 2019-09-26                              618 - Parent withdraw  
5 2021-09-24  618 - Not eligible for Part B – Exit with no r...  
6 2020-10-15                              618 - Parent withdraw  

[5 rows x 38 columns]

In [10]:
dcdf = eligdf[eligdf['Init. Elig. Category'] == 'Diagnosed Condition']

In [33]:
print(dcdf)

      District  Child ID Child Status  Child Phase Init. IFSP Due Date  \
3           ET    446841     Inactive  Eligibility          2018-11-23   
16          ET    451526     Inactive         IFSP          2019-01-27   
20          ET    459059     Inactive         IFSP          2019-05-09   
23          ET    450729     Inactive         IFSP          2019-01-18   
24          ET    450731     Inactive         IFSP          2019-01-18   
...        ...       ...          ...          ...                 ...   
67510       UC    499521     Inactive         IFSP          2021-09-30   
67543       UC    471413     Inactive         IFSP          2021-09-03   
67555       UC    478966     Inactive         IFSP          2021-11-20   
67556       UC    478967     Inactive         IFSP          2021-11-20   
67608       UC    511641     Inactive  Eligibility          2021-11-22   

      IFSP Due Date IFSP Late Reason Active Ref. Date  \
3               NaT              NaN       2018-10-09 

In [12]:
dedcdf = eligdf[eligdf['Init. Elig. Category'] == 'Diagnosed Condition, Developmental Evaluation']

In [32]:
print(dedcdf)

      District  Child ID Child Status Child Phase Init. IFSP Due Date  \
1           ET    431729     Inactive        IFSP          2018-10-26   
10          ET    427633     Inactive        IFSP          2018-11-08   
32          ET    445289     Inactive        IFSP          2018-11-03   
33          ET    448009     Inactive        IFSP          2018-12-08   
43          ET    438450     Inactive        IFSP          2019-03-30   
...        ...       ...          ...         ...                 ...   
67497       UC    510438       Active        IFSP          2021-10-22   
67501       UC    528249       Active        IFSP          2022-06-20   
67520       UC    518770     Inactive        IFSP          2022-02-24   
67588       UC    506474     Inactive        IFSP          2021-08-27   
67605       UC    509553     Inactive        IFSP          2021-10-03   

      IFSP Due Date IFSP Late Reason Active Ref. Date  \
1        2018-10-26              NaN       2018-09-11   
10       

5298 children

In [20]:
dedcdfsuccess = dedcdf[dedcdf['Exit Reason'] == '618 - Completion of IFSP/ no longer requires services']

In [31]:
print(dedcdfsuccess)

      District  Child ID Child Status Child Phase Init. IFSP Due Date  \
363         ET    458721     Inactive        IFSP          2019-05-04   
1372        ET    459631     Inactive        IFSP          2019-05-16   
7212        GN    441639     Inactive        IFSP          2018-09-13   
7448        MD    447901     Inactive        IFSP          2018-12-07   
13114       SE    461491     Inactive        IFSP          2019-06-10   
13563       SE    441336     Inactive        IFSP          2018-09-08   
17265       ET    430392     Inactive        IFSP          2020-02-22   
17994       ET    481331     Inactive        IFSP          2020-05-30   
19984       FT    478771     Inactive        IFSP          2020-04-10   
20761       FT    482297     Inactive        IFSP          2020-06-21   
25409       MD    480987     Inactive        IFSP          2020-05-22   
27711       SC    466833     Inactive        IFSP          2019-09-12   
32365       ET    499131     Inactive        IFSP  

In [26]:
print(dedcdfsuccess)

      District  Child ID Child Status Child Phase Init. IFSP Due Date  \
363         ET    458721     Inactive        IFSP          2019-05-04   
1372        ET    459631     Inactive        IFSP          2019-05-16   
7212        GN    441639     Inactive        IFSP          2018-09-13   
7448        MD    447901     Inactive        IFSP          2018-12-07   
13114       SE    461491     Inactive        IFSP          2019-06-10   
13563       SE    441336     Inactive        IFSP          2018-09-08   
17265       ET    430392     Inactive        IFSP          2020-02-22   
17994       ET    481331     Inactive        IFSP          2020-05-30   
19984       FT    478771     Inactive        IFSP          2020-04-10   
20761       FT    482297     Inactive        IFSP          2020-06-21   
25409       MD    480987     Inactive        IFSP          2020-05-22   
27711       SC    466833     Inactive        IFSP          2019-09-12   
32365       ET    499131     Inactive        IFSP  

In [27]:
dedfsuccess = dedf[dedf['Exit Reason'] == '618 - Completion of IFSP/ no longer requires services']

In [28]:
print(dedfsuccess)

      District  Child ID Child Status Child Phase Init. IFSP Due Date  \
343         ET    459947     Inactive        IFSP          2019-05-19   
353         ET    447200     Inactive        IFSP          2018-11-29   
358         ET    451925     Inactive        IFSP          2019-02-02   
470         ET    450907     Inactive        IFSP          2019-01-20   
511         ET    453134     Inactive        IFSP          2019-02-24   
...        ...       ...          ...         ...                 ...   
64644       SE    510636     Inactive        IFSP          2021-10-24   
64654       SE    506900     Inactive        IFSP          2021-08-23   
65741       SW    497885     Inactive        IFSP          2022-02-26   
66473       UC    507410     Inactive        IFSP          2021-08-30   
66791       UC    514812     Inactive        IFSP          2022-01-01   

      IFSP Due Date IFSP Late Reason Active Ref. Date  \
343      2019-05-15              NaN       2019-04-04   
353      

In [29]:
dcdfsuccess = dcdf[dcdf['Exit Reason'] == '618 - Completion of IFSP/ no longer requires services']

In [30]:
print(dcdfsuccess)

      District  Child ID Child Status Child Phase Init. IFSP Due Date  \
316         ET    445965     Inactive        IFSP          2019-06-21   
329         ET    447837     Inactive        IFSP          2018-12-06   
352         ET    450385     Inactive        IFSP          2019-01-12   
360         ET    440228     Inactive        IFSP          2018-08-24   
604         ET    459388     Inactive        IFSP          2019-05-12   
622         ET    410542     Inactive        IFSP          2019-05-30   
1653        ET    457606     Inactive        IFSP          2019-04-21   
2623        ET    444507     Inactive        IFSP          2018-10-25   
2678        ET    451829     Inactive        IFSP          2019-02-01   
2679        ET    446227     Inactive        IFSP          2018-11-16   
2724        ET    455749     Inactive        IFSP          2019-03-29   
2764        ET    443568     Inactive        IFSP          2018-10-11   
2925        FT    431889     Inactive        IFSP  

In [88]:
exitcountde = dedf['Exit Reason'].value_counts()
exitcountde

618 - Part B eligibility not determined                                 5550
618 - Parent withdraw                                                   5277
618 - Part B eligible                                                   4560
618 - Attempts to contact unsuccessful                                  1923
Parent decline                                                          1753
Unable to contact                                                       1134
618 - Moved out of state                                                 760
618 - Not eligible for Part B – Exit with referral to other programs     758
618 - Not eligible for Part B – Exit with no referrals                   590
No meeting occurred                                                      250
618 - Completion of IFSP/ no longer requires services                    218
618 - Deceased                                                            23
Referral less than 45 days                                                22

In [89]:
exitcountdc = dcdf['Exit Reason'].value_counts()
exitcountdc

618 - Parent withdraw                                                   910
Parent decline                                                          842
618 - Part B eligibility not determined                                 550
618 - Part B eligible                                                   347
618 - Attempts to contact unsuccessful                                  343
Unable to contact                                                       331
618 - Moved out of state                                                122
618 - Not eligible for Part B – Exit with referral to other programs     94
No meeting occurred                                                      69
618 - Not eligible for Part B – Exit with no referrals                   57
618 - Completion of IFSP/ no longer requires services                    38
618 - Deceased                                                           11
Referral less than 45 days                                                4
Name: Exit R

In [90]:
exitcountdedc = dedcdf['Exit Reason'].value_counts()
exitcountdedc

618 - Parent withdraw                                                   923
618 - Part B eligibility not determined                                 615
618 - Part B eligible                                                   600
618 - Attempts to contact unsuccessful                                  455
Parent decline                                                          314
Unable to contact                                                       268
618 - Moved out of state                                                170
618 - Not eligible for Part B – Exit with referral to other programs     86
618 - Not eligible for Part B – Exit with no referrals                   57
No meeting occurred                                                      57
618 - Deceased                                                           28
618 - Completion of IFSP/ no longer requires services                    22
Referral less than 45 days                                                5
Name: Exit R

In [91]:
exitcountdedc.iloc[11]

22

In [92]:
sum(exitcountdedc)

3600

In [93]:
exitcountdedc.iloc[11]/sum(exitcountdedc) *100

0.6111111111111112

.61% of children who have an initial eligibility category of Diagnosed Condition, Developmental Evaluation complete IFSP/ no longer require services.

In [97]:
exitcountde.iloc[10]/sum(exitcountde) * 100

0.9553442306849556

.96% of children who have an initial eligibility category of Developmental Evaluation complete IFSP/ no longer require services.

In [96]:
exitcountde.iloc[10]

218

In [99]:
exitcountdc.iloc[10]/sum(exitcountdc) *100

1.0220548682087143

1.02% of children who have an initial eligibility category of Diagnosed Condition complete IFSP/ no longer require services.